pennylane/pulse/rydberg_machine.py

```diff
@@ -0,0 +1,237 @@
+        # Dictionary containing the information about the local driving fields
+        self._local_drives = {"rabi": [], "detunings": [], "phases": [], "wires": []}
+        # List of tuples containing the information about the global driving field
+        self._global_drive = None  # (rabi frequency, detuning, phase)
+
+    @property
+    def rydberg_interaction(self) -> Sum:
+        r"""Returns the hamiltonian of the interaction of the ensemble of atoms due to the Rydberg
+        blockade:
+
+        .. math::
+
+            \sum_{i<j} V_{ij} n_i n_j
+
+        where :math:`n_i` corresponds to the projector on the Rydberg state of the atom :math:`i`, and
+        :math:`V_{ij}` is the van der Waals potential:
+
+        .. math::
+
+            V_{ij} = \frac{C_6}{R_{ij}^6}
+
+        where :math:`R_{ij}` is the distance between the atoms :math:`i` and :math:`j`, and :math:`C_6`
+        is the Rydberg interaction constant, which can be set during initialization, and defaults
+        to :math:`862690 \times 2\pi MHz \times \mu m^6`.
+
+        Args:
+            qubit_positions (list): list of coordinates of the Rydberg atoms
+            separation (float): separation between the Rydberg atoms (in meters)
+            wires (list): List of wires containing the wire values for all the atoms. This list should
+                have the same length as ``qubit_positions``.
+
+        Returns:
+            Sum: hamiltonian of the interaction
+        """
+        if self._rydberg_interaction is None:
+
+            def rydberg_projector(wire: int) -> SProd:
+                """Returns the projector into the Rydberg state for the given wire.
+
+                Args:
+                    wire (int): _description_
+
+                Returns:
+                    SProd: projector into the Rydberg state
+                """
+                return (1 - qml.PauliZ(wire)) / 2
+
+            coeffs = []
+            ops = []
+            for idx, (pos1, wire1) in enumerate(zip(self.coordinates[1:], self.wires[1:])):
+                for pos2, wire2 in zip(self.coordinates[: idx + 1], self.wires[: idx + 1]):
+                    atom_distance = np.linalg.norm(qml.math.array(pos2) - pos1)
+                    Vij = self.interaction_coeff / (
+                        abs(atom_distance) ** 6
+                    )  # van der Waals potential
+                    coeffs.append(Vij)
+                    ops.append(qml.prod(rydberg_projector(wire1), rydberg_projector(wire2)))
+
+            self._rydberg_interaction = qml.dot(coeffs, ops)
+
+        return self._rydberg_interaction
+
+    def local_drive(self, rabi: list, detunings: list, phases: list, wires: list):
+        """Apply ``N = len(rabi)``  local driving laser fields with the given rabi frequencies,
+        detunings and phases acting on the given wires.
+
+        Args:
+            rabi (list): list of Rabi frequency values (in MHz) of each driving laser field
```

Comment:
I don't think we want these to be lists of values. For the ParametrizedHamiltonian, they should be either a constant, or a callable, so that we end up with either fixed or parametrized terms.

Reply:
Sorry, I misread this the first time through. I think we need to make clear that the each value on the list can be a constant or a callable.

Reply:
Updated docstring 👍